### DISCUSSION TITLE
About ROI size in the sliding_window_inference method

### DISCUSSION THREAD

**Discussion by saruarlive** [Project-MONAI/MONAI#1517] [Q&A]
Hi @Nic-Ma et al,

In sliding_window_inference, what strategy is followed to find ROI from the input image?

Is ROI is selected from the foreground voxel area only?  or Is the ROI selected randomly?
I have trained a model using foreground voxels (CropForeground) only. I want to do inference without cropping/resizing the validation image. How can I achieve it using sliding_window_inference class, doing inference on the foreground voxels without losing dimension/spacing/spatial location of the input image?

Looking forward to hearing from you.

**Comment by saruarlive** [comment]
Hi @Nic-Ma, @rijobro @wyli any comments related to my query!

**Comment by Nic-Ma** [comment]
Hi @saruarlive ,

Sorry for delayed response.
`SlidingWindow` scans the whole input image with predefined `window size` and `overlap ratio`, then generate a output image which has exact same size the input size.
I think if you used `CropForground` transform during training, it's OK to not use it in validation, you can have a try first.

Thanks.
